packages/toolpad-app/src/toolpad/AppEditor/PageEditor/ComponentCatalog/ComponentCatalog.tsx

```diff
@@ -98,6 +115,20 @@ export default function ComponentCatalog({ className }: ComponentCatalogProps) {
     [],
   );
 
+  const handleSearchFilter = React.useCallback(
+    (event: React.ChangeEvent<HTMLInputElement>) => {
+      const { value } = event.target;
+      if (!value) {
+        setToolpadComponentList(toolpadComponents);
+        return;
+      }
+      const regex = new RegExp(value.split('').join('.*'), 'i');
+      const newlist = toolpadComponents.filter(([componentName]) => regex.test(componentName));
```

Comment:
One thing that I'd like to add to the [component definition](https://github.com/mui/mui-toolpad/blob/a8760a4ea5bbf4043d3fc8f0c10d2a01cf6c602e/packages/toolpad-components/src/Autocomplete.tsx#L123) is `synonyms: string[]` property that we can use to declare extra names this filter should match on. e.g. the select component should also appear when typing "combobox" and "dropdown"

Reply:
![image](https://github.com/mui/mui-toolpad/assets/27261522/e09cc066-c9fe-4f57-9e03-b63b79eb2542)
right ?

I just want to make sure your ideas again.

Reply:
Ah yes it can go there indeed 👍

Reply:
> Ah yes it can go there indeed 👍

Do you want to preset some components?
if not.
I'm going to set a empty array there.

Reply:
We could start with:

```
Autocomplete: combobox, select, dropdown
Button: click, action
Chart: graph, bar chart, pie chart, line chart, plot
Checkbox: switch
DataGrid: table
DatePicker: time
Image: picture
List: repeat
Metric: value, number, output
Paper: surface
Select: combobox, dropdown
Text: markdown, link, output
TextField: input, field
```

Reply:
ok i done